forte/processors/data_augment/ranking_augment_processor.py

```diff
@@ -0,0 +1,182 @@
+        if self.config.aug_document.lower() == 'true':
+            packs = {}
+            for doc_id in input_pack.pack_names:
+                if doc_id == query_pack_name:
+                    continue
+
+                aug_doc_pack: DataPack = self._process_doc_pack(input_pack.get_pack(doc_id))
+                # Todo: Does doc_id have to be int type to get elastic search works?
+                packs['aug_' + doc_id] = aug_doc_pack
+
+            input_pack.update_pack(packs)
+
+    def _process_query_pack(self, pack: DataPack) -> DataPack:
+        r"""
+        This function process a single query datapack with an augmenter.
+        It processes the query text by appending with augmented text.
+        :param pack: a query datapack with original texts.
+        :return: a query datapack with augmented texts.
+        """
+        data_pack: DataPack = DataPack()
+        if len(pack.text) == 0:
+            return data_pack
+
+        query_text = pack.text
+        sent_texts = nltk.sent_tokenize(query_text) # query text is usually one sentence.
+
+        pack_text: str = ""
+        for sent in sent_texts:
+            sent_text: str = self.generate_augmented_sentences(sent)
+            pack_text += " " + sent_text if len(pack_text) > 0 else sent_text
+
+        new_query_text = query_text + pack_text
+        data_pack.set_text(new_query_text)  # new_query_text = ori_query_text + aug_texts
+        return data_pack
+
+    def _process_doc_pack(self, pack: DataPack) -> DataPack:
+        r"""
+        This function process a single document datapack with an augmenter.
+        It processes the document text (and annotation, if any).
+        Note that for MSMacro dataset, the document datapack only has Document annotation and pack text defined.
+        :param pack: a document datapack with original texts.
+        :return: a document datapack with augmented texts.
+        """
+        data_pack: DataPack = DataPack()
+        if len(pack.text) == 0:
+            return data_pack
+
+        if len(pack.get(Sentence)):
+            sent_texts = pack.get(Sentence)
+        else:
+            doc_texts = pack.text
+            sent_texts = nltk.sent_tokenize(doc_texts)
+
+        pack_text: str = ""
+        for sentt in sent_texts:
+            sent = sentt.text
+            sent_text: str = self.generate_augmented_sentence(sent)
+
+            # Build the pack text and sentence annotation.
+            start_index: int = len(pack_text)
+            pack_text += " " + sent_text if len(pack_text) > 0 else sent_text
+            if "Sentence" in self.configs.augment_entries:
+                Sentence(data_pack, start_index, len(pack_text))
+
+        # Build the Document annotation
+        if "Document" in self.configs.augment_entries:
+            Document(data_pack, 0, len(pack_text))
+
+        data_pack.set_text(pack_text)
+        return data_pack
+
+    def generate_augmented_sent(self, sent_text: str) -> str:
```

Comment:
These code are almost the same as the dictionary-based replacement augmenter. Why are we reimplement the same code twice?